src/HomeBlaze/HomeBlaze.Components/Widget.cs

```diff
@@ -0,0 +1,58 @@
+using HomeBlaze.Abstractions;
+using HomeBlaze.Abstractions.Attributes;
+using HomeBlaze.Services;
+using Namotion.Interceptor;
+using Namotion.Interceptor.Attributes;
+
+namespace HomeBlaze.Components;
+
+/// <summary>
+/// A subject that references another subject by path and renders its widget component.
+/// Enables embedding widgets by reference in markdown instead of defining them inline.
+/// </summary>
+[InterceptorSubject]
+public partial class Widget : ITitleProvider, IConfigurableSubject
+{
+    private readonly SubjectPathResolver _pathResolver;
+
+    /// <summary>
+    /// Path to the subject to render.
+    /// Supports "Root.Children[folder].Children[file.json]" for absolute paths from root.
+    /// </summary>
+    [Configuration]
+    public partial string Path { get; set; }
```

Comment:
here we should add a new attribute [SubjectPath] or [SubjectReference] so later when we move subjects, we know where we need to update paths